### PR TITLE
Update title

### PR DESCRIPTION
**Pull request by mim24sd**
In this pull request we want to :
1. Change edit icon to submit and change it back.
2. Change input to title and change it back. (when title changed)
3. handle error when title input is empty.
4. show `updatedAt` when task get updated.

### CODE REVIEW

File: assests/javascript/index.js [hunk 1]
```diff
@@ -68,24 +68,39 @@ async function getAllTasks() {
...
-              <i class="fa-solid fa-pen-to-square edit-task-icon"></i>
+              <div id="title-box-${id}" class="title-box">
+                <p class="task-title">${text}</p>
+              </div>
+              <i class="fa-solid fa-check-circle hidden" id="submit-task-icon-${id}"></i>
+              <i class="fa-solid fa-pen-to-square edit-task-icon" id="edit-task-icon-${id}"></i>
               <i class="fa-solid fa-xmark delete-task-icon"></i>
 
-              <p class="tooltiptext">${formatDate(createdAt)}</p>
+              <div class="tooltip-text" id="tooltip-${id}">${tooltipText(createdAt, updatedAt)}</div>
```
Comment (blacksrc) [inline assests/javascript/index.js]:
```suggestion
              <div class="tooltip-text" id="tooltip-${id}">${getTooltipText(createdAt, updatedAt)}</div>
```

File: assests/javascript/index.js [hunk 2]
```diff
@@ -68,24 +68,39 @@ async function getAllTasks() {
...
   handleTaskEvents();
   classifyTasks(tasks);
 }
 
+function tooltipText(createdAt, updatedAt) {
+  let tooltip = `<p>${formatDate(createdAt)}</p>`;
+
+  if (createdAt !== updatedAt) {
+    tooltip = `<p>Created at : ${formatDate(createdAt)}</p>
+               <p>Updated at : ${formatDate(updatedAt)}</p>`
```
Comment (blacksrc) [inline assests/javascript/index.js]:
```suggestion
    tooltip += `<p>Updated at: ${formatDate(updatedAt)}</p>`
```

File: assests/javascript/index.js [hunk 3]
```diff
@@ -68,24 +68,39 @@ async function getAllTasks() {
...
+              <div class="tooltip-text" id="tooltip-${id}">${tooltipText(createdAt, updatedAt)}</div>
             </li>`;
   }).join("");
 
   handleTaskEvents();
   classifyTasks(tasks);
 }
 
+function tooltipText(createdAt, updatedAt) {
+  let tooltip = `<p>${formatDate(createdAt)}</p>`;
```
Comment (blacksrc) [inline assests/javascript/index.js]:
```suggestion
  let tooltip = `<p>Created at: ${formatDate(createdAt)}</p>`;
```

File: assests/javascript/index.js [hunk 4]
```diff
@@ -136,6 +151,125 @@ async function updateTaskStatus(id, isDone) {
...
+  const editTaskIcons = Array.from(document.getElementsByClassName("edit-task-icon"));
+
+  editTaskIcons.forEach((editTaskIcon) => {
+    editTaskIcon.addEventListener("click", (event) => {
+      const id = event.target.parentNode.id;
+      const titleBox = document.getElementById(`title-box-${id}`);
+      const taskTitle = titleBox.innerText;
+
+      convertEditIconToSubmitIcon(id);
+      convertTaskTitleToInput(id, titleBox, taskTitle);
```
Comment (blacksrc) [inline assests/javascript/index.js]: there is no need to pass `titleBox` as an argument, right? you can just select buy id inside? Or am I missing something?

File: assests/javascript/index.js [hunk 5]
```diff
@@ -136,6 +151,125 @@ async function updateTaskStatus(id, isDone) {
...
+function convertTaskTitleToInput(id, titleBox, taskTitle) {
+  titleBox.innerHTML = `<input class="edit-task-title" id="edit-title-${id}" type="text" value="${taskTitle}"/>
+                        <p id="edit-error-${id}" class="edit-title-error"></p>`;
+}
+
+function editTaskTitle(id, taskTitle) {
+  const submitIcon = document.getElementById(`submit-task-icon-${id}`);
+
+  submitIcon.addEventListener("click", function (event) {
+    const newTaskTitle = document.querySelector('input[type=text]').value;
```
Comment (blacksrc) [inline assests/javascript/index.js]: The selector is very generic. Will select all inputs in the page.

File: assests/javascript/index.js [hunk 6]
```diff
@@ -136,6 +151,125 @@ async function updateTaskStatus(id, isDone) {
...
+    const newTaskTitleIsEmpty = !(newTaskTitle.trim());
+
+    if (taskTitle !== newTaskTitle && newTaskTitleIsEmpty === false) {
+      updateTitle(id, newTaskTitle);
+    }
+    else if (newTaskTitleIsEmpty === true) {
+      showEditEmptyInputError(id);
+    }
```
Comment (blacksrc) [inline assests/javascript/index.js]: Maybe you can simplify this logic, test it before pushing.
```suggestion
    const newTaskTitle = document.querySelector('input[type=text]').value.trim();
    
    if (!newTaskTitle) {
       showEditEmptyInputError(id);
    }
    
    if (taskTitle !== newTaskTitle ) {
      updateTitle(id, newTaskTitle);
    } 
```

File: assests/javascript/index.js [hunk 7]
```diff
@@ -136,6 +151,125 @@ async function updateTaskStatus(id, isDone) {
...
+
+  editTaskInput.classList.add("error");
+
+  editTitleError.innerText = "Add title!";
+}
+
+async function updateTitle(id, text) {
+  try {
+    const body = JSON.stringify({ text });
+    const headers = { "Content-Type": "application/json" };
```
Comment (blacksrc) [inline assests/javascript/index.js]: you have this already. no?

File: assests/javascript/index.js [hunk 8]
```diff
@@ -136,6 +151,125 @@ async function updateTaskStatus(id, isDone) {
...
+
+    const response = await fetch(`${baseUrl}/todos/${id}`, {
+      method: "PATCH",
+      body,
+      headers,
+    });
+
+    const { data } = await response.json();
+    updateTaskBox(data)
```
Comment (blacksrc) [inline assests/javascript/index.js]: This is ok and also fine. But maybe for the sake of simplicity. In any case of update you could just re-render the task list. Again, this is fine.

File: assests/javascript/index.js (hunk 1 above)
Comment (mim24sd) [reply assests/javascript/index.js]: doneeee

Comment (mim24sd) [reply assests/javascript/index.js]: doneeee

File: assests/javascript/index.js (hunk 3 above)
Comment (mim24sd) [reply assests/javascript/index.js]: done

File: assests/javascript/index.js (hunk 2 above)
Comment (mim24sd) [reply assests/javascript/index.js]: done

File: assests/javascript/index.js (hunk 7 above)
Comment (mim24sd) [reply assests/javascript/index.js]: yessss

File: assests/javascript/index.js (hunk 4 above)
Comment (mim24sd) [reply assests/javascript/index.js:164]: I will check it with you

File: assests/javascript/index.js (hunk 5 above)
Comment (mim24sd) [reply assests/javascript/index.js]: I will replace it by `className`

File: assests/javascript/index.js (hunk 6 above)
Comment (mim24sd) [reply assests/javascript/index.js]: thankkkks

File: assests/javascript/index.js (hunk 8 above)
Comment (mim24sd) [reply assests/javascript/index.js]: I will ask you more about this

File: assests/javascript/index.js (hunk 4 above)
Comment (mim24sd) [reply assests/javascript/index.js]: You are right
I just wanted to avoid repeating.

File: assests/javascript/index.js [hunk 9]
```diff
@@ -112,12 +123,12 @@ async function deleteTask(id) {
 
 function handleStatus() {
   const checkboxes = Array.from(document.getElementsByClassName("task-check-box"));
```
Comment (blacksrc) [inline assests/javascript/index.js:125]: use `querySelector` instead.

**Pull request merged**